### ISSUE TITLE
About negative samples

### ISSUE THREAD

**Issue by yunzi-94** [soledad921/ATISE#3]
In your paper  TeRo, you have said "In this paper, we use the same loss function as the negative sampling loss proposed in (Sun et al.,2019) for optimizing our model." 

But in your code (shared), I can only find a simple random negative sampling method [#](https://github.com/soledad921/ATISE/blob/master/Train.py)    ( Line 80-140 in Train.py)

How do you think about the adversarial negative sampling method used in RotatE (Sun et al.,2019)?

**Comment by soledad921** [comment]
The adversarial negative sampling is actually implemented in the loss function, Please check the log_rank_loss functions defined in the model.py (line 120-123). For sure, the adversarial negative sampling is proven to be helpful for boosting (T)KGE models' performances in many cases including ours.

**Comment by yunzi-94** [comment]
Thanks for your reply, and it really helps me a lot. 
But I still have a question: without the adversarial negative sampling, can we still make a conclusion how efficiently TeRo can behave? (I have seen the comparisons between with/without adversarial negative sampling in RotatE paper (Sun et al.,2019).)
If you are interested in this query, we can discuss it further.

**Comment by soledad921** [comment]
> Thanks for your reply, and it really helps me a lot.
> But I still have a question: without the adversarial negative sampling, can we still make a conclusion how efficiently TeRo can behave? (I have seen the comparisons between with/without adversarial negative sampling in RotatE paper (Sun et al.,2019).)
> If you are interested in this query, we can discuss it further.

Generally, the adversarial negative sampling can improve 1-3 MRR points in cases of TeRo and RotatE.
